researcher_ids/RI_Claims_V1.md

```diff
@@ -585,106 +592,255 @@ the GA4GH DURI committee.
+    Visa if it can confirm that the conditions have been met.
+
+-   Although it is RECOMMENDED to always implement full condition checking
+    capabilities as described in this section, if a Claim Clearinghouse will be
+    deployed for a more limited purpose where it is not expected to ever receive
+    Passport Visas with conditions, then such a Claim Clearinghouse MAY choose to
+    not implement full condition checking. However, even in this case it MUST
+    still check for the presence of the `conditions` field on Passport Visa
+    Objects and reject (ignore) any Passport Visas that contain a non-empty
+    `conditions` field value. Likewise if not all condition matching algorithms
+    are implemented, it MUST reject any Passport Visas that contain patterns
+    that are not supported.
 
--   In the process of finding a matching Passport Visa for a condition, a
-    Passport Clearinghouse MUST ignore Passport Visa Objects that also have a
-    condition. This will avoid deep nesting of condition evaluation (i.e.
-    avoid condition loops, stack overflows, etc).
+-   Format:
 
--   [Passport Visa Fields](#passport-visa-fields) that are not specified
-    in the condition are not required to match (i.e. any value will be accepted
-    within that field and optional fields need not be present).
+    ```
+    "conditions": [
+      [
+        { // Condition clause 1
+          "type": "<passport-visa-type>",
+          "<passport-visa-object-field1>": "<match-type>:<match-value>",
+          "<passport-visa-object-field2>": "<match-type>:<match-value>",
+          ...
+        }, // AND
+        { // Condition clause 2
+          ...
+        }
+      ], // OR
+      [
+        { // Condition clause 3
+          "type": "<passport-visa-type>",
+          "<passport-visa-object-field>": "<match-type>:<match-value>",
+          ...
+        }
+      ],
+      ...
+    ]
+    ```
 
--   Format:
+-   The `conditions` value is a two-nested-lists structure in Disjunctive
+    Normal Form:
+    
+    -   The outer level list is a set of OR clauses
+    
+    -   The inner level list is a set of AND clauses that contain "Condition
+        Clauses"
+
+    -   A Condition Clause MUST specify a "type" field with a value as a
+        Passport Visa Type plus it MUST include at least one other field with a
+        name that matches a valid Passport Visa Object field name.
+      
+    -   The values of Condition Clause fields MUST have a string prefix followed
+        by a colon and then string suffix, except for "type" where it MUST be
+        assumed to be "const" and is not specified.
+        
+        -   If prefix is "const", then suffix MUST use case sensitive full string
+            matching.
+
+        -   If prefix is "pattern", then suffix MUST use full string [Pattern
+            Matching](#pattern-matching) to match input.
+
+        -   If prefix is "split_pattern", then the full suffix MUST use full
```

Comment:
I do not like that the split_pattern uses a single predefined character as the point for splitting. I would either remove the split_pattern altogether to simplify implementations, or allow it to specify arbitrary character for splitting. Just my two cents.

Reply:
Adding extra text around this use case to precondition its use where a Passport Visa Type designs the splitting, such as it has been done for LinkedIdentities. We can look at a separate PR to introduce variables or other means to specify the separator, but as now written this capability is essentially saying that if you are planning on using multi-valued strings, use URI-encoded substrings with semi-colon separators like what LinkedIdentities does.